ledger/block/src/verify.rs

```diff
@@ -508,72 +510,97 @@ impl<N: Network> Block<N> {
         transactions: &Transactions<N>,
         aborted_transaction_ids: &[N::TransactionID],
     ) -> Result<()> {
-        // Prepare an iterator over the solution IDs.
-        let mut solutions = solutions.as_ref().map(|s| s.deref()).into_iter().flatten().peekable();
-        // Prepare an iterator over the unconfirmed transaction IDs.
-        let unconfirmed_transaction_ids = cfg_iter!(transactions)
-            .map(|confirmed| confirmed.to_unconfirmed_transaction_id())
-            .collect::<Result<Vec<_>>>()?;
-        let mut unconfirmed_transaction_ids = unconfirmed_transaction_ids.iter().peekable();
-
-        // Initialize a list of already seen transmission IDs.
-        let mut seen_transmission_ids = HashSet::new();
-
-        // Initialize a list of aborted or already-existing solution IDs.
-        let mut aborted_or_existing_solution_ids = Vec::new();
-        // Initialize a list of aborted or already-existing transaction IDs.
-        let mut aborted_or_existing_transaction_ids = Vec::new();
-
-        // Iterate over the transmission IDs.
-        for transmission_id in subdag.transmission_ids() {
-            // If the transmission ID has already been seen, then continue.
-            if !seen_transmission_ids.insert(transmission_id) {
-                continue;
+        match subdag {
+            Subdag::Compact { .. } => {
+                // get length of solutions.
+                let solutions_len = solutions.as_ref().map(|c| c.len()).unwrap_or(0);
+                // get length of transactions.
+                let transactions_len = transactions.len() + aborted_transaction_ids.len();
+                // Get bitset of unique solution indices.
+                let solution_indices = subdag.solution_indices(solutions_len)?;
+                // Get bitset of unique transaction indices.
+                let tx_indices = subdag.transaction_indices(transactions_len)?;
+                // Ensure the Subdag contains all solution indices.
+                ensure!(solution_indices.len() == solutions_len, "Subdag does not contain all tx indices.");
+                for i in 0..2usize.pow(solutions_len as u32) {
+                    ensure!(solution_indices.contains(i), "Subdag does not contain all tx indices.");
+                }
+                // Ensure the Subdag contains all transaction indices.
+                ensure!(tx_indices.len() == transactions_len, "Subdag does not contain all tx indices.");
+                for i in 0..2usize.pow(transactions_len as u32) {
```

Comment:
Ditto on the above,. We can theoretically have `2^TRANSACTIONS_DEPTH` transactions in a block, so would in theory have to iterate and call `contains` `2^(2^TRANSACTIONS_DEPTH)` times

Reply:
Fixed https://github.com/AleoHQ/snarkVM/commit/a7ed56b2f32df6d0ba20515535ce31133bf24d69